tree/ntuple/doc/BinaryFormatSpecification.md

```diff
@@ -684,6 +685,64 @@ The entry range allows for finding the right page list for random access request
+
+1. it cannot have linked Attribute RNTuples itself;
+2. the Alias columns sections, both in its header and footer, must be empty (i.e. none of the Attribute Set RNTuple's
+   Fields can be Projected Fields);
+3. none of its fields may have a structural role of 0x04 (i.e. it must not contain a ROOT streamer object);
+
+The payload of each list element is a Record Frame containing the following fields:
+
+- a 32-bit unsigned integer containing the _Attribute Schema Version_. This is split into a _Major_ (16 LSB) and a
+  _Minor_ (16 MSB) version. The Schema Version is described below;
+- a 32-bit unsigned integer containing the uncompressed size of the Anchor;
+- a Locator pointing to the Attribute RNTuple's Anchor;
+- a String containing the Attribute Set's name. All linked Attribute Sets must have a non-empty, distinct name.
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|     Schema Version Major     |     Schema Version Minor       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|             Attribute Anchor Uncompressed Size                |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   Attribute Anchor Locator                    |
+.                             ...                               .
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                      Attribute Set Name                       |
+.                             ...                               .
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+##### Attribute Schema Version
+Each Attribute Set is created with a user-defined Model. This Model is not used directly by the underlying Attribute
+Set RNTuple, but it is augmented with internal fields used to store additional data that serve to associate each
+entry in the Attribute Set with those in the main RNTuple.
+
+The Attribute Schema Version describes the internal schema of the linked Attribute Set RNTuple.
+A change in Major version number indicates a breaking, non-forward-compatible change in the schema: readers should
+refuse reading an Attribute Set whose Major Schema Version is unknown.
+A change in Minor version number indicates the presence of optional additional fields in the schema: readers should
+still be able to read the Attribute Set as before, ignoring any new field.
+
+The current Attribute Schema Version is **1.0**. It has the following fields (in the following order):
+  1. `_rangeStart` (type `NTupleSize_t`): the start of the range that each Attribute Entry refers to;
+  2. `_rangeLen` (type `NTupleSize_t`): the length of the range that each Attribute Entry refers to.
```

Comment:
`NTupleSize_t` is a `typedef` that I believe we are not treating specially during type normalization. In any case, it's not defined so far in the spec, I think it would be better to resolve this to `std::uint64_t`.